pyccel/ast/cupyext.py

```diff
@@ -0,0 +1,488 @@
+
+    def __init__(self, shape, fill_value, dtype=None, order='C'):
+
+        # Convert shape to PythonTuple
+        shape = process_shape(False, shape)
+        # If there is no dtype, extract it from fill_value
+        # TODO: must get dtype from an annotated node
+        if dtype is None:
+            dtype = fill_value.dtype
+            precision = get_final_precision(fill_value)
+        else:
+            dtype, precision = process_dtype(dtype)
+
+        # Cast fill_value to correct type
+        if fill_value:
+            if fill_value.dtype != dtype or get_final_precision(fill_value) != precision:
+                cast_func = DtypePrecisionToCastFunction[dtype.name][precision]
+                fill_value = cast_func(fill_value)
+        self._shape = shape
+        self._rank  = len(self._shape)
+        self._dtype = dtype
+        self._order = CupyNewArray._process_order(self._rank, order)
+        self._precision = precision
+
+        super().__init__(fill_value)
+
+    #--------------------------------------------------------------------------
+    @property
+    def fill_value(self):
+        return self._args[0]
+
+#==============================================================================
+class CupyAutoFill(CupyFull):
+    """ Abstract class for all classes which inherit from CupyFull but
+        the fill_value is implicitly specified
+    """
+    __slots__ = ()
+    def __init__(self, shape, dtype='float', order='C'):
+        if not dtype:
+            raise TypeError("Data type must be provided")
+        super().__init__(shape, Nil(), dtype, order)
+
+#==============================================================================
+class CupyEmpty(CupyAutoFill):
+    """ Represents a call to cupy.empty for code generation.
+    """
+    __slots__ = ()
+    name = 'empty'
+
+    def __init__(self, shape, dtype='float', order='C'):
+        if dtype in NativeNumeric:
+            precision = default_precision[str_dtype(dtype)]
+            dtype = DtypePrecisionToCastFunction[dtype.name][precision]
+        super().__init__(shape, dtype, order)
+    @property
+    def fill_value(self):
+        return None
+
+
+#==============================================================================
+class CupyZeros(CupyAutoFill):
+    """ Represents a call to cupy.zeros for code generation.
+    """
+    __slots__ = ()
+    name = 'zeros'
+    @property
+    def fill_value(self):
+        dtype = self.dtype
+        if isinstance(dtype, NativeInteger):
+            value = LiteralInteger(0, precision = self.precision)
+        elif isinstance(dtype, NativeFloat):
+            value = LiteralFloat(0, precision = self.precision)
+        elif isinstance(dtype, NativeComplex):
+            value = LiteralComplex(0., 0., precision = self.precision)
+        elif isinstance(dtype, NativeBool):
+            value = LiteralFalse(precision = self.precision)
+        else:
+            raise TypeError('Unknown type')
```

Comment:
Why not use `convert_to_literal` ?

```suggestion
        value = convert_to_literal(0, dtype, self.precision)
```